adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -629,21 +816,34 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
     optPartitions: Option[Int] = None)(
       implicit tTag: ClassTag[T], xTag: ClassTag[X]): GenomicRDD[(T, Option[X]), Z] = {
 
-    val (partitionSize, endSequences) = joinPartitionSizeAndSequences(optPartitions,
-      genomicRdd)
+    // did the user provide a set partition count?
+    // if no, take the max partition count from our rdds
+    val partitions = optPartitions.getOrElse(Seq(rdd.partitions.length,
+      genomicRdd.rdd.partitions.length).max)
+
+    val leftRdd: GenomicRDD[T, U] = if (sorted) {
+      if (partitions != rdd.partitions.length) evenlyRepartition(partitions) else this
+    } else {
+      repartitionAndSort(partitions)
+    }
+
+    val preparedRdds = prepareForShuffleRegionJoin(leftRdd, genomicRdd)
+    val leftRddReadyToJoin = preparedRdds._1
+    val rightRddReadyToJoin = preparedRdds._2
+    assert(leftRddReadyToJoin.partitions.length == rightRddReadyToJoin.partitions.length)
```

Comment:
This assert is the last thing we do in `prepareForShuffleRegionJoin`, right? I would eliminate it here.